packages/web-components/@sfx/product/test/unit/product.test.ts

```diff
@@ -0,0 +1,199 @@
+import { TemplateResult, html } from 'lit-element';
+import { Base } from '@sfx/base';
+
+import { expect, stub, spy } from '../utils';
+
+import Product from '../../src/product';
+import Variant from '../../src/variant';
+
+describe('Product Component', () => {
+  let component;
+
+  beforeEach(() => {
+    component = new Product();
+  });
+
+  describe('constructor', () => {
+    it('should extend the Base class', () => {
+      expect(component).to.be.an.instanceof(Base);
+    });
+
+    describe('product property', () => {
+      it('should have a default value of "full"', () => {
+        expect(component.product).to.exist;
+      });
+    });
+  });
+
+  describe('connectedCallback', () => {
+    it('should call its super connectedCallback', () => {
+      const connectedCallback = stub(Base.prototype, 'connectedCallback');
+
+      component.connectedCallback();
+      expect(connectedCallback).to.have.been.called;
+    });
+
+    it('should add an eventListener to the window', () => {
+      const addEventListener = spy(window, 'addEventListener');
+
+      component.connectedCallback();
+      expect(addEventListener).to.have.been.calledWith(
+        'sfx::change_product_variant',
+        component.updateVariant
+      );
+    });
+  });
+
+  describe('disconnectedCallback', () => {
+    it('should call its super disconnectedCallback', () => {
+      const disconnectedCallback = stub(Base.prototype, 'disconnectedCallback');
+
+      component.disconnectedCallback();
+      expect(disconnectedCallback).to.have.been.called;
+    });
+
+    it('should remove eventListener from the window', () => {
+      const removeEventListener = spy(window, 'removeEventListener');
+
+      component.disconnectedCallback();
+      expect(removeEventListener).to.have.been.calledWith(
```

Comment:
These tests should be updated to follow the convention we've been using: Arrange Act Assert (all separated by newlines). In this case, we just need a newline above the `expect` line here.